src/sentry/search/events/builder/discover.py

```diff
@@ -860,6 +861,9 @@ def resolve_orderby(self, orderby: list[str] | str | None) -> list[OrderBy]:
                 or isinstance(resolved_orderby, AliasedExpression)
             ):
                 bare_orderby = resolved_orderby.alias
+                if resolved_orderby.alias in orderby_no_columns:
+                    validated.append(OrderBy(resolved_orderby, direction))
+                    break
```

Comment:
This is also a bit of a hack. I'd lean towards just passing the column you're ordering by into the select so you can see what it's ordering on.